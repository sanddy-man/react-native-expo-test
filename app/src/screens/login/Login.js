import React, { Component } from "react";
import { View, ImageBackground, Image } from "react-native";
import { Text } from "native-base";
import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
import { Constants } from "../../config";
import styles from "./styles";
import { Forms } from "../../components";
import { SecureStore } from 'expo';
import { getlang } from "../../config/constants";
import multilingual from "../../config/multilingual";

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: ""
    };
  }

  componentDidMount() {
    getlang().then(lang => {
      this.setState({lang})
    })
  }

  login() {
    fetch(Constants.API_ROOT + "/login", {
      method: "POST",
      body: JSON.stringify({
        email: this.state.email,
        password: this.state.password
      }),
    }).then((response) => response.json())
    .then((responseJson) => {
      if (responseJson.code != 200){
        alert(responseJson.message);
        return;
      }

      SecureStore.setItemAsync("jwt", responseJson.token);
      this.resetState();
      this.props.navigation.navigate("_dashboard");
    })
    .catch((error) => {
      console.log(error);
    });
  }

  navigateToRegister() {
    this.resetState();
    this.props.navigation.navigate("_register");
  }

  resetState() {
    this.setState({
      email: "",
      password: ""
    });
  }

  render() {
    const { lang } = this.state
    return (
      <ImageBackground source={require("../../assets/images/background.png")} style={styles.backgroundImage} >
        <KeyboardAwareScrollView
          enableOnAndroid
          enableAutomaticScroll
          keyboardOpeningTime={0}
        >
          <View style={styles.content}>
            <View style={styles.centered}>
              <View style={styles.imageContainer}>
                <Image
                  source={require("../../assets/images/nicu_logo.png")}
                  style={styles.image}
                />
              </View>

              <Text style={styles.introText}>{multilingual.INTRO_TEXT_1[lang]}</Text>
              <Text style={styles.introText}>{multilingual.INTRO_TEXT_2[lang]}</Text>

              <View style={styles.bottom}>
                <Forms.LoginForm
                  email={this.state.email}
                  onEmailChange={text =>
                    this.setState({ email: text })
                  }
                  password={this.state.password}
                  onPasswordChange={text => this.setState({ password: text })}
                  onCreateAccountPress={() => this.navigateToRegister()}
                  onSubmitPress={() => this.login()}
                  lang={lang}
                />
              </View>
            </View>
          </View>
        </KeyboardAwareScrollView>
      </ImageBackground>
    );
  }
}

export default Login;
