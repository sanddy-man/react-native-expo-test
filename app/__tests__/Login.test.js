import React from "react";
import Login from "../src/screens/login/Login";

import renderer from "react-test-renderer";

it("renders correctly", () => {
  const rendered = renderer.create(<Login />).toJSON();
  expect(rendered).toMatchSnapshot();
});
