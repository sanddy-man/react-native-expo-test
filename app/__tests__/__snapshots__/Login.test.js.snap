// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#FFF",
      },
    ]
  }
  virtual={undefined}
>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 15,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          Object {
            "backgroundColor": "#FFF",
            "borderBottomColor": "#FFF",
          },
        ]
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            },
            Object {
              "flex": 3,
            },
          ]
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "600",
                "textAlign": "center",
              },
              Object {
                "color": "#000",
                "fontFamily": "LatoBold",
                "fontSize": 18,
              },
            ]
          }
          virtual={undefined}
        >
          Login
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
        virtual={undefined}
      />
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={undefined}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={0}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "padding": 16,
          }
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontFamily": "LatoBlack",
                "fontSize": 20,
              }
            }
          >
            WELCOME
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#5F5F5F",
                "fontFamily": "AvenirMedium",
                "fontSize": 16,
                "marginTop": 5,
              }
            }
          >
            Sign in to your House Rental account.
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <View
            style={Object {}}
            virtual={undefined}
          >
            <View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": 1,
                    "borderColor": "#D9D5DC",
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderTopWidth": 0,
                    "flexDirection": "row",
                    "marginLeft": 0,
                    "marginTop": 15,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "left": 0,
                      "opacity": 1,
                      "paddingBottom": undefined,
                      "paddingTop": undefined,
                      "position": "absolute",
                      "right": 0,
                      "top": 18,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#575757",
                          "fontSize": 17,
                          "left": 0,
                          "paddingRight": 5,
                          "top": 6,
                        },
                        Object {
                          "color": "#5C6979",
                          "fontFamily": "AvenirMedium",
                          "fontSize": 16,
                          "paddingTop": 4,
                        },
                      ]
                    }
                    virtual={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "fontSize": 17,
                          },
                          Object {
                            "color": "#5C6979",
                            "fontFamily": "AvenirMedium",
                            "fontSize": 16,
                            "paddingTop": 4,
                          },
                        ]
                      }
                      virtual={undefined}
                    >
                      Mobile Number
                    </Text>
                  </Text>
                </View>
                <TextInput
                  allowFontScaling={true}
                  disabled={undefined}
                  editable={true}
                  keyboardType="numeric"
                  maxLength={undefined}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholderTextColor="#575757"
                  returnKeyType="next"
                  secureTextEntry={undefined}
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "flex": 1,
                        "fontSize": 17,
                        "height": 50,
                        "lineHeight": 24,
                        "marginLeft": null,
                        "paddingLeft": null,
                        "paddingRight": 5,
                        "top": 10,
                      },
                      Object {
                        "color": "#000",
                        "fontFamily": "AvenirMedium",
                        "fontSize": 17,
                      },
                    ]
                  }
                  underlineColorAndroid="rgba(0,0,0,0)"
                  value=""
                  virtual={undefined}
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "marginTop": 20,
                }
              }
            />
            <View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": 1,
                    "borderColor": "#D9D5DC",
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderTopWidth": 0,
                    "flexDirection": "row",
                    "marginLeft": 0,
                    "marginTop": 15,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "left": 0,
                      "opacity": 1,
                      "paddingBottom": undefined,
                      "paddingTop": undefined,
                      "position": "absolute",
                      "right": 0,
                      "top": 18,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#575757",
                          "fontSize": 17,
                          "left": 0,
                          "paddingRight": 5,
                          "top": 6,
                        },
                        Object {
                          "color": "#5C6979",
                          "fontFamily": "AvenirMedium",
                          "fontSize": 16,
                          "paddingTop": 4,
                        },
                      ]
                    }
                    virtual={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "fontSize": 17,
                          },
                          Object {
                            "color": "#5C6979",
                            "fontFamily": "AvenirMedium",
                            "fontSize": 16,
                            "paddingTop": 4,
                          },
                        ]
                      }
                      virtual={undefined}
                    >
                      Password
                    </Text>
                  </Text>
                </View>
                <TextInput
                  allowFontScaling={true}
                  disabled={undefined}
                  editable={true}
                  keyboardType="default"
                  maxLength={undefined}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholderTextColor="#575757"
                  returnKeyType="next"
                  secureTextEntry={true}
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "flex": 1,
                        "fontSize": 17,
                        "height": 50,
                        "lineHeight": 24,
                        "marginLeft": null,
                        "paddingLeft": null,
                        "paddingRight": 5,
                        "top": 10,
                      },
                      Object {
                        "color": "#000",
                        "fontFamily": "AvenirMedium",
                        "fontSize": 17,
                      },
                    ]
                  }
                  underlineColorAndroid="rgba(0,0,0,0)"
                  value=""
                  virtual={undefined}
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "marginTop": 20,
                }
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#000",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 3,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#FFF",
                    "fontFamily": "LatoRegular",
                    "fontSize": 16,
                  }
                }
              >
                Login
              </Text>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "AvenirMedium",
                  "fontSize": 16,
                  "marginTop": 20,
                  "textAlign": "center",
                }
              }
            >
              Tap here to create an account
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
